src/codegen/infer.jl

```diff
@@ -84,26 +81,20 @@ end
 
 function inferVariable(scope, expr::Expr)
 	if @capture(expr, a_::b_{t__})
-		push!(scope.locals, a)
-		return WGPUVariable(a, eval(b), Generic, nothing, ) # TODO t is ignored
+		scope.locals[a] = Base.eval(b)
```

Comment:
This needs more changes. globals should hold original variable and locals should hold reference.
Also globals should be named differently to avoid confusion during scope resolution. This will help define local scope variable different from outer scope variable with same name.